runner/sidecar/sink/s3/s3.go

```diff
@@ -44,10 +45,23 @@ func New(ctx context.Context, secretInterface v1.SecretInterface, x dfv1.S3Sink)
 		}
 		secretAccessKey = string(secret.Data[x.Credentials.SecretAccessKey.Key])
 	}
+	var sessionToken string
```

Comment:
Same here, using pointer to simplify the logic.

Reply:
we need to revisit the AWS Credentials altogether. I kept it similar to the existing patterns.